Feature/gui

vuorioi opened the pull request:
I'm little bit on the fence about placing the `.qml` files in the same directory with the Python sources, but for the time being this will do. The QML source also needs some documenting.

Currently only the platform screen is implemented. There are two classes: one for running on an embedded device (`PlatformScreen`) and one for running on development machine (`VirtualPlatformScreen`).

Qt5 runtime libs are needed for running.

Closes #5

vuorioi committed:
Changed license to GPL v3

vuorioi committed:
Added PyQt5 to requirements.txt

vuorioi committed:
Added a GUI package for managing the screens

The current implementation contains only a platform screen for
displaying the current lifters informations etc.

tlammi commented on kratos/gui/screen.py:
```diff
@@ -0,0 +1,49 @@
+    # pylint: disable=too-few-public-methods
+    # This is a base class and has (at the moment) only limited functionality
+
+    _argv = []
```
Should this be a parameter to constructor instead of an attribute?

vuorioi replied:
Yeah, a parameter to the ctor is probably less confusing.

tlammi commented on kratos/gui/platform_view.qml:
```diff
@@ -0,0 +1,293 @@
+                            target: applicationData
+
+                            onJudgingSet: {
+                                leftJudgeDecision.color = judging[0] ? "green" : "red"
```
green -> white

vuorioi committed:
Changed good-lift color from green to white

vuorioi committed:
Changed argument passing to use constructor parameters

Added an additional method to PlatformScreen for setting up the
PlatformData model and connecting it to the QML view.

tlammi commented on kratos/gui/screen.py:
```diff
@@ -0,0 +1,53 @@
+    # pylint: disable=too-few-public-methods
+    # This is a base class and has (at the moment) only limited functionality
+
+    def __init__(self, argv=[]):
```
Mutable default arguments should be avoided: https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

```Python
    def __init__(self, argv=None):
        """
        Ctor
        :param argv: Arguments passed to the underlying QGuiApplication
        """
        argv = argv or []
```
In the above snippet argv is argv if it's evaluated as True and [] otherwise. None evaluates as False in if-statements.

vuorioi committed:
Unfucked the mutable default argument
